File: Modules/HumanResource/PayrollSettings/PayslipPaidOneTimeAllowance/PayslipPaidOneTimeAllowanceForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.PayrollSettings.Forms;

[FormScript("PayrollSettings.PayslipPaidOneTimeAllowance")]
[BasedOnRow(typeof(PayslipPaidOneTimeAllowanceRow), CheckNames = true)]
public class PayslipPaidOneTimeAllowanceForm
{
    public int AllowanceId { get; set; }
    public int PayslipId { get; set; }
    public int IsActive { get; set; }
    public DateTime InsertDate { get; set; }
    public DateTime UpdateDate { get; set; }
    public DateTime DeleteDate { get; set; }
    public int InsertUserId { get; set; }
    public int UpdateUserId { get; set; }
    public int DeleteUserId { get; set; }
}

File: Modules/ServerTypes/PayrollSettings/PayslipPaidOneTimeAllowanceForm.ts
import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PayslipPaidOneTimeAllowanceForm {
    AllowanceId: IntegerEditor;
    PayslipId: IntegerEditor;
    IsActive: IntegerEditor;
    InsertDate: DateEditor;
    UpdateDate: DateEditor;
    DeleteDate: DateEditor;
    InsertUserId: IntegerEditor;
    UpdateUserId: IntegerEditor;
    DeleteUserId: IntegerEditor;
}

export class PayslipPaidOneTimeAllowanceForm extends PrefixedContext {
    static formKey = 'PayrollSettings.PayslipPaidOneTimeAllowance';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PayslipPaidOneTimeAllowanceForm.init)  {
            PayslipPaidOneTimeAllowanceForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(PayslipPaidOneTimeAllowanceForm, [
                'AllowanceId', w0,
                'PayslipId', w0,
                'IsActive', w0,
                'InsertDate', w1,
                'UpdateDate', w1,
                'DeleteDate', w1,
                'InsertUserId', w0,
                'UpdateUserId', w0,
                'DeleteUserId', w0
            ]);
        }
    }
}